modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -1835,6 +1835,210 @@ class DeConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
+        CV_Assert(ngroups > 0 && inpCn % ngroups == 0 && numOutput % ngroups == 0);
+
+        outputs.resize(inputs.size(), shape(&dims[0], dims.size()));
+        return false;
+    }
+
+    virtual void finalize(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr)
+    {
+        CV_Assert(!blobs.empty());
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        if (padMode == "VALID")
+        {
+            int dst[] = {0, 0, 0};
+            pads = DictValue::arrayInt(dst, 3);
+        }
+        else if (padMode == "SAME")
+        {
+            std::vector<int> dst(6);
+            for (int i = 0; i < pads.size() / 2; i++) {
+                dst[i] = dst[i + 3] = std::max(0, (outputs[0].size[i + 2] - 1) * strides.get<int>(i) +
+                                        (dilations.get<int>(i) * (kernel.get<int>(i) - 1) + 1) - inputs[0].size[i + 2] ) / 2;
+            }
+            pads = DictValue::arrayInt(&dst[0], 6);
+        }
+    }
+
+    virtual Ptr<BackendNode> initInfEngine(const std::vector<Ptr<BackendWrapper> > &inputs) CV_OVERRIDE
+    {
+#ifdef HAVE_INF_ENGINE
+        InferenceEngine::DataPtr input = infEngineDataNode(inputs[0]);
+        CV_Assert(input->dims.size() == 5);
+
+        const int inpCn = input->dims[3];  // NOTE: input->dims are reversed (whdcn)
+        const int outCn = blobs[0].size[0];
+        const int inpGroupCn = blobs[0].size[1];
+        const int group = inpCn / inpGroupCn;
+        weightsMat = blobs[0];
+
+        auto ieWeights = wrapToInfEngineBlob(blobs[0], InferenceEngine::Layout::NCDHW);
+
+        if (weightsMat.isContinuous())
+        {
+            Mat fusedWeights = weightsMat.reshape(1, blobs[0].dims, blobs[0].size);
+            ieWeights = wrapToInfEngineBlob(fusedWeights, InferenceEngine::Layout::NCDHW);
+        }
+        else
+        {
+            ieWeights = InferenceEngine::make_shared_blob<float>(
+                                InferenceEngine::Precision::FP32, InferenceEngine::Layout::NCDHW,
+                                ieWeights->dims());
+            ieWeights->allocate();
+
+            Mat newWeights = infEngineBlobToMat(ieWeights).reshape(1, outCn);
+            Mat fusedWeights = weightsMat.colRange(0, newWeights.cols);
+            fusedWeights.copyTo(newWeights);
+        }
+
+        InferenceEngine::Blob::Ptr ieBiases;
+        if (hasBias())
+        {
+            biasvec = blobs[1];
+            Mat biasesMat({outCn}, CV_32F, &biasvec[0]);
+            ieBiases = wrapToInfEngineBlob(biasesMat, {(size_t)outCn}, InferenceEngine::Layout::C);
+        }
+
+#if INF_ENGINE_VER_MAJOR_GE(INF_ENGINE_RELEASE_2018R5)
+        InferenceEngine::Builder::ConvolutionLayer ieLayer(name);
+
+        ieLayer.setKernel({(size_t)kernel.get<int>(0), (size_t)kernel.get<int>(1), (size_t)kernel.get<int>(2)});
```

Comment:
Please experiment with merging `Convolution3D` into the `Convolution` layer. IE backend receives vectors as convolution's hyperparameters so it might looks like:

```
// std::vector<size_t> kernel: height+width for Convolution and depth+height+width for Convolution3D
ieLayer.setKernel(kernel);
```